libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/MapboxTripStarterExtra.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.core.trip
+
+import androidx.annotation.StringDef
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+
+/**
+ * Specifies a trip type for the [MapboxTripStarter].
+ */
+internal object MapboxTripStarterExtra {
```

Comment:
Since this is internal, I believe we can have an `enum class` instead. This way compiler will tell us if we forget to support a new trip type in `MapboxTripStarter`.

Reply:
Or maybe even `sealed class`:
```kotlin
internal sealed class MapboxTripStarterState {
    data class MapMapching(val locationPermissionGranted: Boolean) : MapboxTripStarterState()
    data class ReplayRoute(val options: ReplayRouteSessionOptions) : MapboxTripStarterState()
}
```
I think this will let you have a single state flow in `MapboxTripStarter`.

Reply:
Totally in support of this. 👍🏻

Reply:
Seems nicer from here, but it ends up with a different behavior.

The `ReplayRouteSessionOptions` are lost when you switch between `MapMatching` and `ReplayRoute`.

Reply:
we can do this though

``` kotlin
internal sealed class MapboxTripStarterType {
    object MapMatching : MapboxTripStarterType()
    object ReplayRoute : MapboxTripStarterType()
}
```

Reply:
Yeah this is what I had in mind, I missed the state as suggested by Vitaly. But the above changes look good to me.

Reply:
You can save the latest `ReplayRouteSessionOptions` in a separate property. And the rest of the code will be simpler, you will have a single flow and won't need `flatMapLatest`. But it is not a big deal, so I leave it up to you.

Reply:
Yeah I still think the `flatMapLatest` will be needed so that changes can take effect. Or there will be conditional statements that update the current state.

We can consider the approach more when adding replay history cause it could look like this. But it can be a future pull request.
```
    data class ReplayHistory(
        val filePath: String,
        val mapper: ReplayHistoryMapper
    ) : MapboxTripStarterType()
```